### PR TITLE
Concise info on creating and using a module without core and base

### PR DESCRIPTION
**Pull request by udohjeremiah**
I had a hard time solving this  - (using `@eval` to write code into the module and then writing a function doc for the new function evaluated using `@eval`).

Thank God for the helpful and timely answers I got from slack and discourse, then I finally knew how to use `@eval` on working with functions on the module without `core` and `base`.

Then finally documenting the function was another series of asking questions and finally the timely answers came. And I could add a function docstring to it.

But the point is, why not just show a quick demo of this in the documentation so others don't have to pass through this stress?

### CODE REVIEW

File: doc/src/manual/modules.md [hunk 1]
```diff
@@ -325,7 +325,30 @@ include(p) = Base.include(Mod, p)
...
-If even `Core` is not wanted, a module that imports nothing and defines no names at all can be defined with `Module(:YourNameHere, false, false)` and code can be evaluated into it with [`@eval`](@ref) or [`Core.eval`](@ref).
+If even `Core` is not wanted, a module that imports nothing and defines no names at all can be defined with `Module(:YourNameHere, false, false)` and code can be evaluated into it with [`@eval`](@ref) or [`Core.eval`](@ref):
+```julia
+julia> arithmetic = Module(:arithmetic, false, false)
+Main.arithmetic
+
+julia> @eval arithmetic add(x, y) = $(+)(x, y)
+add (generic function with 1 method)
+
+julia> @doc """
```
Comment (Keno) [inline doc/src/manual/modules.md:336]: `@doc` here seems somewhat random since that isn't introduced anywhere else on this page.

Comment (udohjeremiah) [reply doc/src/manual/modules.md:336]: If that's the case, then should the `@doc` be put into `@ref` so users can click on it to get information on what the `@doc` says?

Comment (udohjeremiah) [reply doc/src/manual/modules.md]: > If that's the case, then should the `@doc` be put into `@ref` so users can click on it to get information on what the `@doc` says?

I've updated the examples. I think its concise now. Take a look @Keno

File: doc/src/manual/modules.md [hunk 2]
```diff
@@ -325,7 +325,30 @@ include(p) = Base.include(Mod, p)
...
+```jldoctest
+julia> arithmetic = Module(:arithmetic, false, false)
+Main.arithmetic
+
+julia> @eval arithmetic add(x, y) = $(+)(x, y)
+add (generic function with 1 method)
+
+julia> """
+           add(x, y)
+
```
Comment (vtjnash) [inline doc/src/manual/modules.md:338]:
```suggestion
       
```

**Comment by KristofferC** [comment]
```
Whitespace check found 1 issues:
--
  | doc/src/manual/modules.md:338 -- trailing whitespace
  | make: *** [Makefile:106: check-whitespace] Error 1
```

**Comment by udohjeremiah** [comment]
> Whitespace check found 1 issues:
> --
>   | doc/src/manual/modules.md:338 -- trailing whitespace
>   | make: *** [Makefile:106: check-whitespace] Error 1

I think its fixed.

**Comment by vtjnash** [comment]
Maybe it shouldn't try to test adding docstrings from the REPL (or add a more trivial one)?

**Comment by udohjeremiah** [comment]
> Maybe it shouldn't try to test adding docstrings from the REPL (or add a more trivial one)?

Yeah I've added a more trivial one which shouldn't throw an error.

**Comment by udohjeremiah** [comment]
> I still don't think the `@doc` and `help?>` examples here make any sense. They're a complete side track to anything being discussed here. It's really not a good idea to just throw in random unrelated concepts in examples.

And  I think they're one of the reasons the whitespace is failing.

I'll remove it now.

**Comment by udohjeremiah** [comment]
@ViralBShah are you looking at this since you're more active in doc PRs.

**Comment by udohjeremiah** [comment]
@ViralBShah I certainly do not know what you mean by **"obvious"** but I feel this discussion 👉#46536  opened here concerning merging stuffs like this is a valid one. You might to check out what I'm talking about.

**Comment by ViralBShah** [comment]
Doc PRs touch many different aspects and need to have reviews from appropriate people. Thus, I can only merge the ones that I am sure about - on others I try to find other reviewers. But yes, developer bandwidth continues to be limited, although we try to do our best to stay on top of open PRs. A lot of people have commit access to the repo, but on a given PR there's often only a few people who can authoritatively merge.

I am happy to be pinged on such PRs to take a look.

**Comment by udohjeremiah** [comment]
OK!

**Pull request merged**